src/StorageFlattening.cpp

```diff
@@ -398,15 +398,11 @@ class FlattenDimensions : public IRMutator {
     Stmt visit(const For *op) override {
         bool old_in_shader = in_shader;
         bool old_in_gpu = in_gpu;
-        if ((op->for_type == ForType::GPUBlock ||
-             op->for_type == ForType::GPUThread) &&
-            op->device_api == DeviceAPI::GLSL) {
-            in_shader = true;
-        }
         if (op->for_type == ForType::GPUBlock ||
             op->for_type == ForType::GPUThread) {
             in_gpu = true;
         }
+        // TODO: in_shader = true is no longer possible, clean up code accordingly
```

Comment:
I believe the branch that adds texture support for cuda might need this

Reply:
Please point me at branch?

Reply:
#5423

Reply:
It doesn't appear to use `in_shader` -- it adds a new clause to preserve the device-api